legate/core/runtime.py

```diff
@@ -358,15 +395,15 @@ def __init__(
         # This is a sanitized list of (region,field_id) pairs that is
         # guaranteed to be ordered across all the shards even with
         # control replication
-        self.free_fields: Deque[tuple[Region, int]] = deque()
+        self.free_fields: Deque[
```

Comment:
I feel `_try_reuse_field` would become much simpler if we had two deques, one for those free of pending detach ops and one for those with them.